src/core/components/dag.js

```diff
@@ -9,6 +9,17 @@ const flattenDeep = require('lodash.flattendeep')
 module.exports = function dag (self) {
   return {
     put: promisify((dagNode, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+      }
+
+      const optionDefaults = {
+        format: 'dag-pb',
+        hashAlg: 'sha2-256'
```

Comment:
@alanshaw you might wanna review those.

Reply:
These should be the same as go-ipfs (https://ipfs.io/docs/commands/#ipfs-dag-put) i.e. dag-cbor and sha2-256

Note also that `cid` can be passed here instead of `format` & `hashAlg` that you need to take into account. See https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/DAG.md#dagput